server/athenian/api/internal/miners/jira/issue.py

```diff
@@ -1344,6 +1344,28 @@ def resolve_resolved(
     return resolved
 
 
+def resolve_acknowledge_time(
+    created: npt.NDArray[np.datetime64],
+    work_began: npt.NDArray[np.datetime64],
+    now: np.datetime64,
+) -> npt.NDArray[np.timedelta64]:
+    """Compute the acknowledge_time for the issues.
+
+    `created` are the `Issue.created.name` values
+    `work_began` are the final values computed by resolve_work_began()
+
+    The acknowledge_time is `work_began` - `created`.
+    For the issues not yet started (so no work_began) it is `now()` - `created`.
+    `acknowledge_time` can never be less than 0.
+    """
+    res = np.empty(len(created), dtype="timedelta64[us]")
+    res = work_began - created
+    no_work_began = res != res
+    res[no_work_began] = now - created[no_work_began]
+    res[res < np.timedelta64(0, "us")] = np.timedelta64(0, "us")
+    return res
```

Comment:
This duplicates `AcknowledgeTimeCalculator` and is actually not fully consistent with it. We'd rather try to refactor the calc by making a static subfunction.

Reply:
They start from a different definition of "effective_work_began", so let's try first to  uniform this:

- for `jira-acknowledge-time` metric
`effective_work_began = fmin(work_began, prs_began) AND status == TODO`

- for `jira-lead-time` metric
`effective_work_began = fmin(work_began, prs_began)`

- for /jira/filter output
`effective_work_began = fmin(work_began, prs_began) AND work_began is not NaT`

I can use third definition for `jira-acknowledge-time` but that would change result of the metric when status is todo and work_began is filled.

Reply:
It is `status != TODO`: `acknowledged[status == TODO] = None` in the code. It is equivalent to checking `work_began == work_began` so the definitions are the same. Let's not touch the lead time though.